src/Features/ParentStatus.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2022 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Features;
+
+use CommonITILActor;
+use CommonITILObject;
+use PendingReason_Item;
+use Session;
+
+/**
+ * ParentStatus
+ *
+ * @since 10.0.4
+ */
+trait ParentStatus
+{
+    public function updateParentStatus(): void
```

Comment:
Changes made to `$this->input` seems to be used only in this new method. Passing values in parameters would make static code analysis easier for IDEs and code analysis tools.

```suggestion
    public function updateParentStatus(CommonITILObject $parentitem, array $input): void
```